enterprise/tailnet/pgcoord_test.go

```diff
@@ -592,8 +592,6 @@ func TestPGCoordinatorDual_Mainline(t *testing.T) {
 	err = client21.recvErr(ctx, t)
 	require.ErrorIs(t, err, io.EOF)
 
-	assertEventuallyNoAgents(ctx, t, store, agent2.id)
```

Comment:
What should happen is that agent2, client22, and client21 should all be set to "LOST" in the database as their connections are torn down, and you should assert that instead.